lib/src/widget/settings/widget/settings_tile.dart

```diff
@@ -24,15 +24,28 @@ class SettingsTile extends StatelessWidget {
       children: rowChildren,
       crossAxisAlignment: CrossAxisAlignment.start,
     );
-    return SizedBox(
-        height: linden.dimen.unit7,
-        child: Padding(
-          padding: EdgeInsets.only(
-            left: linden.dimen.unit2,
-            right: linden.dimen.unit0_5,
-          ),
-          child: row,
-        ));
+
+    final content = SizedBox(
+      height: linden.dimen.unit7,
+      child: Padding(
+        padding: EdgeInsets.only(
+          left: linden.dimen.unit2,
+          right: linden.dimen.unit0_5,
+        ),
+        child: row,
+      ),
+    );
+
+    final embedInButton =
+        data.action?.tapArea == SettingsTileActionTapArea.content;
+    return embedInButton
+        ? AppGhostButton(
+            key: data.action?.key,
+            onPressed: data.action?.onPressed,
+            contentPadding: EdgeInsets.zero,
+            child: content,
+          )
+        : content;
```

Comment:
❓ from code it looks like if the `TapArea` == `trailin` - it will be not clickable at all. Not sure it is correct

Reply:
Yes, it will be - the button for cases other than `SettingsTileActionIcon` is further down in the widget tree.